src/http.rs

```diff
@@ -184,46 +183,23 @@ impl<B> fmt::Debug for HttpClient<B> {
     }
 }
 
-#[derive(Deserialize, Serialize, Clone, Debug)]
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq)]
 #[serde(deny_unknown_fields, rename_all = "snake_case", tag = "strategy")]
 pub enum Auth {
     Basic { user: String, password: String },
     Bearer { token: String },
 }
```

Comment:
It occurs to me that almost everywhere we use `enum Auth`, it is as `Option<Auth>`. Am I seeing that right? I appreciate the composability, but it prevents us from writing `fn merge(self: &Option<Self>). Would it make more sense for `None` to be one of the cases here, so that `fn merge_auth_config` could become something like the following.
```rust
impl Auth {
    fn or<'a>(&'a self, other: &'a Auth) -> &'a Self {
        match self {
            None => other,
            _ => self,
        }
    }
}
```
An alternative path would be to create a newtype `MaybeAuth(Option<Auth>)` and then you could write methods on that type.

Reply:
Yes, authentication is all optional at the moment. I don't think we should add `None` to `Auth` because some sink might want to use it as a hard requirement in the future, so `MaybeAuth` would be preferable.

There would be only one method for `MaybeAuth` at the moment, I'm not sure if it is worth introducing the type yet. I will take some time to try it out tomorrow.

Reply:
@bruceg I'm using a `MaybeAuth` at the moment, I prefer `choose_one` over `or`, let me know what you think!